lib/std/json/test.zig

```diff
@@ -29,8 +29,7 @@ fn ok(s: []const u8) !void {
 fn err(s: []const u8) void {
     testing.expect(!json.validate(s));
 
-    testNonStreaming(s) catch return;
-    testing.expect(false);
+    testing.expect(std.meta.isError(testNonStreaming(s)));
```

Comment:
```suggestion
    testing.expectError(error.WhateverTheErrorIs, testNonStreaming(s));
```

Reply:
on second thought this is a pretty weird looking function and your code is definitely an improvement. happy to merge as is :+1: